dotnet/sample-apps/aws-sdk/wrapper/SampleApps/AwsSdkSample/Function.cs

```diff
@@ -0,0 +1,48 @@
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.Core;
+using Amazon.S3;
+using OpenTelemetry;
+using OpenTelemetry.Contrib.Instrumentation.AWSLambda.Implementation;
+using OpenTelemetry.Trace;
+using System;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]
+
+namespace AwsSdkSample
+{
+    public class Function
+    {
+        public static TracerProvider tracerProvider;
+
+        static Function()
+        {
+            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+
+            tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddAWSInstrumentation()
+                .AddOtlpExporter()
+                .AddAWSLambdaConfigurations()
+                .Build();
+        }
+
+        // use AwsSdkSample::AwsSdkSample.Function::TracingFunctionHandler as input Lambda handler instead
+        public APIGatewayProxyResponse TracingFunctionHandler(APIGatewayProxyRequest request, ILambdaContext context)
+        {
+            return AWSLambdaWrapper.Trace(tracerProvider, FunctionHandler, request, context);
+        }
+
+        /// <summary>
+        /// A simple function that takes a APIGatewayProxyRequest and returns a APIGatewayProxyResponse
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public APIGatewayProxyResponse FunctionHandler(APIGatewayProxyRequest request, ILambdaContext context)
+        {
+            var S3Client = new AmazonS3Client();
+            _ = S3Client.ListBucketsAsync().Result;
```

Comment:
Is this an async call ? If .Result is a blocking operation it is OK, else change to sync call.

Reply:
It's ok if Result blocks but yeah better to use idiomatic pattern for sync calls, or alternatively use `await` as idiomatic pattern of async call

Reply:
@wangzlei @anuraaga AWS .NET SDK for `AmazonS3Client` only has `async` APIs. See: https://github.com/aws/aws-sdk-net/blob/475822dec5e87954b7a47ac65995714ae1f1b115/sdk/src/Services/S3/Generated/_netstandard/AmazonS3Client.cs. We can add `.Result` for the `sync` use case.